### PR TITLE
Kotlin Cleanup: Changed to scope function in DeckPickerTest

### PR DESCRIPTION
**Pull request by alyoshenka**
## Purpose / Description
Fix `@KotlinCleanup` tag for better code styling.

## Fixes
Related to #10489 

## Approach
~This code changes from using a variable representation to a scoped function. This helps to enhance readability. The variable is referenced within scope by `it`.~
Originally, the intention was to switch from a variable representation to a scoped function. Further suggestions have instead favored switching to `useHashMapOf` to make things more Kotlin-like.
`getResourceString` is favored over `targetContext.getString` for its simplicity.

## How Has This Been Tested?
This has been tested both as an individual test file and as part of the entire gradle test suite, including testing with incorrect values to ensure they are caught:
- `./gradlew AnkiDroid:testPlayDebugUnitTest --tests com.ichi2.anki.DeckPickerTest.verifyCodeMessages`
- `./gradlew AnkiDroid:testPlayDebugUnitTest --tests com.ichi2.anki.DeckPickerTest`
- `./gradlew jacocoTestReport`

## Learning (optional, can help others)
[This](https://kotlinlang.org/docs/scope-functions.html) is Kotlin's Scope Function reference.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- ~[ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)~
- ~[ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)~

### CODE REVIEW

**Comment by BrayanDSO** [comment]
As I think that the cleanup purpose was to remove the repetitions of `codeResponsePairs`, I'd do (untested)
```kotlin
@Test
    fun verifyCodeMessages() {
        val context = targetContext
        val codeResponsePairs = hashMapOf(
            407 to context.getString(R.string.sync_error_407_proxy_required),
            409 to context.getString(R.string.sync_error_409),
            413 to context.getString(R.string.sync_error_413_collection_size),
            500 to context.getString(R.string.sync_error_500_unknown),
            501 to context.getString(R.string.sync_error_501_upgrade_required),
            502 to context.getString(R.string.sync_error_502_maintenance),
            503 to context.getString(R.string.sync_too_busy),
            504 to context.getString(R.string.sync_error_504_gateway_timeout)
        )
        ActivityScenario.launch(DeckPicker::class.java).use { scenario ->
            scenario.onActivity { deckPicker: DeckPicker ->
                for ((key, value) in codeResponsePairs) {
                    assertEquals(deckPicker.rewriteError(key), value)
                }
            }
        }
    }
```

Pinging @lukstbit, as you're the cleanup author and may have a stronger opinion on that

**Comment by mikehardy** [comment]
Meta note: github will close an issue if you say `Fixes #<number>` (or "closes") but if you just say `Related #<number>` you get the link without the automation, so still some nice cross-reference links between the two but it won't close

**Comment by lukstbit** [comment]
I was probably thinking about an `apply {}` to encapsulate the `HashMap` initialization code. But, as usually, we should also strive to make the code more kotlin like so @BrayanDSO 's suggestion to use the platform `hashMapOf` is much nicer.

@alyoshenka Thank you for the contribution, I would ask you to use @BrayanDSO 's  proposed code as it makes it slightly more readable.

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt [hunk 1]
```diff
@@ -59,17 +58,17 @@ class DeckPickerTest : RobolectricTest() {
...
-        val context = targetContext
-        codeResponsePairs[407] = context.getString(R.string.sync_error_407_proxy_required)
-        codeResponsePairs[409] = context.getString(R.string.sync_error_409)
-        codeResponsePairs[413] = context.getString(R.string.sync_error_413_collection_size)
-        codeResponsePairs[500] = context.getString(R.string.sync_error_500_unknown)
-        codeResponsePairs[501] = context.getString(R.string.sync_error_501_upgrade_required)
-        codeResponsePairs[502] = context.getString(R.string.sync_error_502_maintenance)
-        codeResponsePairs[503] = context.getString(R.string.sync_too_busy)
-        codeResponsePairs[504] = context.getString(R.string.sync_error_504_gateway_timeout)
+        // getResourceString() is the wrapper for targetContext.getString(resource)
```
[the remark this answers is not in the document]
Comment (alyoshenka) [reply AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt]: Thanks for the suggestion, it takes a moment to get accustomed to new coding styles and I appreciate the help!

**Review by BrayanDSO — APPROVED** [review]
Looks good to me. Thanks for the contribution!

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt]: > it takes a moment to get accustomed to new coding styles and I appreciate the help!

I know this feeling. Especially here, with more than 300 contributors with very different past experience, code styles and personalities. But time solves all of the issues

**Review by lukstbit — APPROVED** [review]
Thanks!

**Pull request merged**